packages/webapp/src/hooks/useSearchHandler.ts

```diff
@@ -13,6 +17,26 @@ export function useSearchHandler<T>(
 ) {
 	items = items || empty
 	useResetFilterOnDataChange(items, onFilter)
+	const { orgId } = useCurrentUser()
+	const location = useLocation()
+
+	const handleTrackEvent = () => {
+		trackEvent({
+			name: 'Search',
+			properties: {
+				'Organization ID': orgId,
+				Page: location?.pathname ?? ''
+			}
+		})
+	}
+
+	const debounceTrackFn = useRef(
```

Comment:
I would love if you could send me where you find that trick, I broke my mind of trying to debounce using `useMemo`...

Reply:
I found it on this article: https://www.carlrippon.com/using-lodash-debounce-with-react-and-ts/